File: task1/GO/main.go
package main

import (
	"fmt"
	"math"
)

func calculateY(x float64) float64 {
	if x >= -5.0 && x <= -3.0 {
		return math.Sqrt(4.0 - math.Pow(x+3.0, 2))
	} else if x > -3.0 && x <= -2.0 {
		return -x - 1.0
	} else if x > -2.0 && x <= -1.0 {
		return math.Sqrt(1.0 - math.Pow(x+2.0, 2))
	} else if x > -1.0 && x <= 1.0 {
		return -0.5*x - 0.5
	} else if x > 1.0 && x <= 2.0 {
		return -1.0 - math.Sqrt(1.0-math.Pow(x-2.0, 2))
	} else if x > 2.0 && x <= 3.0 {
		return -x
	} else if x > 3.0 && x <= 5.0 {
		return -3.0 + math.Sqrt(4.0-math.Pow(x-5.0, 2))
	} else {
		return math.NaN()
	}
}

func main() {
	xStart := -5.0
	xEnd := 5.0
	dx := 0.5

	fmt.Println("-------------------------")
	// % — сигнализирует о начале команды
	// s — сокращение от string (строка). Говорит, что мы вставляем текст.
	fmt.Printf("|%8s |%12s |\n", "X", "Y")
	fmt.Println("-------------------------")

	for x := xStart; x <= xEnd+dx/10.0; x += dx {
		y := calculateY(x)

		// %8.2f означает ширину 8 символов и 2 знака после запятой
		fmt.Printf("|%8.2f |", x)

		if math.IsNaN(y) {
			fmt.Printf("%12s |\n", "N/A")
		} else {
			fmt.Printf("%12.4f |\n", y)
		}
	}
	fmt.Println("-------------------------")
}

File: task1/main.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

// Функция для вычисления значения Y в зависимости от X
double calculateY(double x) {
    if (x >= -5.0 && x <= -3.0) {
        return sqrt(4.0 - pow(x + 3.0, 2));
    } 
    else if (x > -3.0 && x <= -2.0) {
        return -x - 1.0;
    } 
    else if (x > -2.0 && x <= -1.0) {
        return sqrt(1.0 - pow(x + 2.0, 2));
    } 
    else if (x > -1.0 && x <= 1.0) {
        return -0.5 * x - 0.5;
    } 
    else if (x > 1.0 && x <= 2.0) {
        return -1.0 - sqrt(1.0 - pow(x - 2.0, 2));
    } 
    else if (x > 2.0 && x <= 3.0) {
        return -x;
    } 
    else if (x > 3.0 && x <= 5.0) {
        return -3.0 + sqrt(4.0 - pow(x - 5.0, 2));
    } 
    else {
        // Если X выходит за пределы графика
        return NAN; 
    }
}

int main() {
    // Исходные данные для полного покрытия графика
    double x_start = -5.0;
    double x_end = 5.0;
    double dx = 0.5;

    // Вывод шапки таблицы
    cout << "---------------------------" << endl;
    cout << "|" << setw(8) << "X" << " |" << setw(12) << "Y" << " |" << endl;
    cout << "---------------------------" << endl;

    // Цикл вычисления и вывода значений
    // + dx/10.0 используется для компенсации погрешности чисел с плавающей точкой
    for (double x = x_start; x <= x_end + dx / 10.0; x += dx) {
        double y = calculateY(x);
        
        cout << "|" << setw(8) << fixed << setprecision(2) << x << " |";
        
        if (isnan(y)) {
            cout << setw(12) << "N/A" << " |" << endl;
        } else {
            cout << setw(12) << fixed << setprecision(4) << y << " |" << endl;
        }
    }
    cout << "---------------------------" << endl;

    return 0;
}